vms/platformvm/warp/benchmarks/benchmarks_test.go

```diff
@@ -0,0 +1,374 @@
+	"github.com/ava-labs/avalanchego/vms/platformvm/signer"
+	"github.com/ava-labs/avalanchego/vms/platformvm/state"
+	"github.com/ava-labs/avalanchego/vms/platformvm/status"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/stretchr/testify/require"
+)
+
+// To profile a specific benchmark, say BenchmarkGetCanonicalValidatorSetByDepth, run:
+// go test -run=^$ -bench ^BenchmarkGetCanonicalValidatorSetByDepth$ github.com/ava-labs/avalanchego/vms/platformvm/warp/benchmarks -cpuprofile=cpuprofile.out
+//
+// To analyze results run:
+// go tool pprof -web cpuprofile.out
+
+func BenchmarkGetCanonicalValidatorSetBySize(b *testing.B) {
+	b.StopTimer()
+	require := require.New(b)
+
+	// Prepare the bench environment (don't time it)
+	subnetID := ids.GenerateTestID()
+	env := newEnvironment(b, subnetID)
+	defer func() {
+		require.NoError(shutdownEnvironment(env))
+	}()
+
+	pChainHeight := uint64(1)
+	numNodes := 10_000
+
+	// Store [numNodes] validators in state. They are all at the same height [pChainHeight]
+	diff, err := state.NewDiff(env.state.GetLastAccepted(), env.blkManager)
+	require.NoError(err)
+	for i := 0; i < numNodes; i++ {
+		nodeID := ids.GenerateTestNodeID()
+
+		// create primary network validator tx
+		addPrimaryValidatorTx, err := benchPrimaryNetworkValidatorTx(env.state, nodeID)
+		require.NoError(err)
+
+		// store corresponding primaryStaker in the diff
+		primaryStaker, err := state.NewCurrentStaker(
+			addPrimaryValidatorTx.ID(),
+			addPrimaryValidatorTx.Unsigned.(*txs.AddPermissionlessValidatorTx),
+			10000, // potential reward
+		)
+		require.NoError(err)
+		diff.PutCurrentValidator(primaryStaker)
+		diff.AddTx(addPrimaryValidatorTx, status.Committed)
+
+		// create subnet validator tx
+		permissionlessValidatorTx, err := benchSubnetValidatorTx(env.state, subnetID, nodeID)
+		require.NoError(err)
+
+		// store corresponding staker in the diff
+		subnetStaker, err := state.NewCurrentStaker(
+			permissionlessValidatorTx.ID(),
+			permissionlessValidatorTx.Unsigned.(*txs.AddPermissionlessValidatorTx),
+			10000, // dummy potential reward
+		)
+		require.NoError(err)
+		diff.PutCurrentValidator(subnetStaker)
+		diff.AddTx(permissionlessValidatorTx, status.Committed)
+	}
+
+	// Add a dummyBlock to update relevant quantities
+	dummyValidatorsBlock, err := blocks.NewBanffStandardBlock(
+		env.state.GetTimestamp(),
+		env.state.GetLastAccepted(),
+		pChainHeight,
+		[]*txs.Tx{},
+	)
+	require.NoError(err)
+
+	// Push block and tx changes to env.state.
+	require.NoError(diff.Apply(env.state))
+	env.state.SetHeight(pChainHeight)
+	env.state.AddStatelessBlock(dummyValidatorsBlock, choices.Accepted)
+	env.state.SetLastAccepted(dummyValidatorsBlock.BlockID)
+	require.NoError(env.state.Commit())
+
+	b.StartTimer() // start testing
+	for _, size := range []int{
```

Comment:
Is this intended to be the size of the validator set being mocked? I only see usage for the test name so I was unsure

Reply:
A leftover. Fixed